pkg/api/handlers/compat/images.go

```diff
@@ -316,7 +319,18 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	w.WriteHeader(http.StatusOK)
+	if pullError != nil {
```

Comment:
First, an unsynchronized concurrent read/write from different goroutines is just invalid per https://go.dev/ref/mem .

And even if it were valid, I’d expect the code to get to this point before the goroutine makes any significant progress (a pull starts with a HTTP round-trip to detect registry features and required authentication, which is certainly going to take longer than just getting to this point), so this should AFAICS be equivalent to `if(false)`.